### PR TITLE
Remove duplicated code in Sqlite ModelDB implementations

### PR DESCRIPTION
**Pull request by rjwebb**
Issue: https://github.com/canvasxyz/canvas/issues/363

This PR removes a lot of duplicated code from `modeldb-sqlite`, `modeldb-sqlite-wasm` and `modeldb-durable-objects`, which are all Sqlite-based ModelDB implementations. This mostly comes by only having one `ModelAPI.ts` file, then passing in a wrapped database object which has a consistent API. I suspect we could also reduce a lot of repetition if we could have a single `ModelDB` class as well. This shared code is exported from `@canvas-js/modeldb/utils`.

To do:
- [x] Add missing dependencies
- [x] Double check everything

## How has this been tested?

- [ ] CI tests pass
- [ ] Tested with example-chat (including login, all signers, and exchanging messages)
- [ ] Tested with `@canvas-js/test-network`: (optional)

## Does this contain any breaking changes to external interfaces?

- [ ] Contract interfaces
- [ ] Core interface
- [ ] CLI
- [ ] Data storage formats, including IndexedDB, SQLite, or filesystem storage (will this break existing apps?)

### CODE REVIEW

File: packages/modeldb/src/utils/AbstractSqliteDB.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import { SqlitePrimitiveValue } from "./encoding.js"
+
+export abstract class AbstractSqliteDB {
+	// exec ? or maybe always prepare
```
Comment (raykyri) [inline packages/modeldb/src/utils/AbstractSqliteDB.ts:4]: comment

File: packages/modeldb/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"extends": "../../tsconfig.base.json",
-	"include": ["src"],
+	"include": ["src", "src/sqlite-utils"],
```
Comment (raykyri) [inline packages/modeldb/tsconfig.json:3]: is this still used

File: packages/modeldb/src/utils/AbstractSqliteDB.ts (hunk 1 above)
Comment (rjwebb) [reply packages/modeldb/src/utils/AbstractSqliteDB.ts]: i think having a method/query class is fine, this was a note to self and should not have been commited!

File: packages/modeldb/tsconfig.json (hunk 2 above)
Comment (rjwebb) [reply packages/modeldb/tsconfig.json]: good catch, i've deleted it

**Pull request closed without merging**